service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/AzureDatabaseConnectedTest.java

```diff
@@ -95,10 +101,81 @@ public void createAndDeleteAzureManagedIdAndDatabase() throws InterruptedExcepti
               assertNotNull(actualDatabase);
             });
 
+    azureDatabaseUtilsRunner.testDatabaseConnect(
+        azureTestUtils.getAzureCloudContext(),
+        workspaceUuid,
+        "default",
+        "test-connect",
+        getDbServerName(),
+        dbResource.getDatabaseName(),
+        uamiResource.getManagedIdentityName());
+
     deleteDatabase(userRequest, dbResource);
     deleteManagedIdentity(userRequest, uamiResource);
   }
 
+  @Test
+  public void createAndDeleteK8sNamespaceWithDatabase() throws InterruptedException {
+    AuthenticatedUserRequest userRequest = userAccessUtils.defaultUserAuthRequest();
+
+    var uamiResource = createManagedIdentity(userRequest);
+    var dbResource = createDatabase(userRequest, uamiResource, null);
+    var k8sResource = createKubernetesNamespace(userRequest, uamiResource, List.of(dbResource));
+
+    // Verify that the resources we created
+    var resourceList = wsmResourceService.enumerateResources(workspaceUuid, null, null, 0, 100);
+    checkForResource(resourceList, uamiResource);
+    checkForResource(resourceList, dbResource);
+    checkForResource(resourceList, k8sResource);
+
+    // wait for azure to sync then make sure the resources actually exist
+    Awaitility.await()
+        .atMost(1, TimeUnit.MINUTES)
+        .pollInterval(5, TimeUnit.SECONDS)
+        .untilAsserted(
+            () -> {
+              var actualUami =
+                  getManagedIdentityFunction()
+                      .apply(
+                          azureTestUtils.getAzureCloudContext().getAzureResourceGroupId(),
+                          uamiResource.getManagedIdentityName());
+              assertNotNull(actualUami);
+
+              var actualDatabase =
+                  getDatabaseFunction()
+                      .apply(
+                          azureTestUtils.getAzureCloudContext().getAzureResourceGroupId(),
+                          dbResource.getDatabaseName());
+              assertNotNull(actualDatabase);
+            });
+
+    azureDatabaseUtilsRunner.testDatabaseConnect(
+        azureTestUtils.getAzureCloudContext(),
+        workspaceUuid,
+        k8sResource.getKubernetesNamespace(),
+        "test-connect",
+        getDbServerName(),
+        dbResource.getDatabaseName(),
+        k8sResource.getKubernetesServiceAccount());
+
+    deleteKubernetesNamespace(userRequest, k8sResource);
```

Comment:
Should these deletions be in a finally in case a failure is hit before this?

Reply:
no, they are part of the test. The @AfterAll function deletes the workspace which cleans everything up